src/lib.rs

```diff
@@ -1,42 +1,22 @@
 #![doc = include_str!("../README.md")]
 #![deny(missing_docs)]
 
-use error::EnvDeserializationError;
-use serde::de::{value::StringDeserializer, DeserializeOwned, IntoDeserializer};
+use serde::de::DeserializeOwned;
 use value::Value;
 
+mod config;
 mod error;
 mod value;
 
-#[derive(Debug, PartialEq)]
-struct Key {
-    original: String,
-    current: String,
-}
-
-impl AsRef<str> for Key {
-    fn as_ref(&self) -> &str {
-        &self.current
-    }
-}
-
-impl Key {
-    fn new(original: String) -> Self {
-        Self {
-            current: original.clone(),
-            original,
-        }
-    }
-}
-
-impl<'de> IntoDeserializer<'de, EnvDeserializationError> for Key {
-    type Deserializer = StringDeserializer<EnvDeserializationError>;
-    fn into_deserializer(self) -> Self::Deserializer {
-        self.current.into_deserializer()
-    }
-}
+pub use config::Config;
+use value::Parser;
 
 /// Whether to use and strip a prefix, and if so which one
+#[derive(Debug, Copy, Clone)]
+#[deprecated(
+    since = "0.3.0",
+    note = "please use `Config` and its `with_prefix` method instead"
+)]
```

Comment:
I felt that deprecating existing functions in `0.3.0` and removing them in `0.4.0`, whenever that is, would make sense; but given I've chosen a breaking change increment then they could be removed altogether. Alternatively they could remain indefinitely as shorthands.